shunting-yard-replicator/src/main/java/com/hotels/shunting/yard/replicator/exec/receiver/ContextFactory.java

```diff
@@ -93,6 +94,7 @@ private String workOutReplicaLocation(String databaseName, String tableName) {
   }
 
   public Context createContext(MetaStoreEvent event) {
+    String circusTrainConfigLocation = PropertyUtils.stringProperty(conf, CT_CONFIG);
```

Comment:
Should we maybe turn this into a File and check that it exists? (uncertain where we should do this but what happens if the file doesn't exist? what will break and when?)

Reply:
The argument is optional now.

Reply:
OK, what happens if you set this but to a path that doesn't exist? Does something throw an error and if so what does it say?

Reply:
I would want SY to fail to start. It will need to happen in commonBeans though as context gets created only on receipt of event. 

how does this look:

```
@Bean
  Configuration baseConfiguration(
      @Value("${instance.workspace}") String workspace,
      @Value("${ct-config:#{null}}") String circusTrainConfigLocation) {
    checkNotNull(workspace, "instance.workspace is required");
    Configuration baseConf = new Configuration();
    baseConf.set(WORKSPACE.key(), workspace);

    if (circusTrainConfigLocation != null) {
      validateCtConfigPath(circusTrainConfigLocation);
      baseConf.set(CT_CONFIG.key(), circusTrainConfigLocation);
    }
    return baseConf;
  }

  private void validateCtConfigPath(String circusTrainConfigLocation) {
    List<String> errors = new ArrayList<>();
    if (StringUtils.isBlank(circusTrainConfigLocation)) {
      errors.add("No config file was specified.");
    } else {
      for (String configFileString : Splitter.on(',').split(circusTrainConfigLocation)) {
        File configFile = new File(configFileString);
        if (!configFile.exists()) {
          errors.add("Config file " + configFileString + " does not exist.");
        } else if (!configFile.isFile()) {
          errors.add("Config file " + configFileString + " is a directory.");
        } else if (!configFile.canRead()) {
          errors.add("Config file " + configFileString + " cannot be read.");
        }
      }
    }
    if (!errors.isEmpty()) {
      throw new ConfigFileValidationException(errors);
    }
  }
```

Reply:
Looks good, you should be able to unit test most of those paths too.

Reply:
Done. I have added a ConfigFileValidator to handle it in couple of places.